src/nvme/tree.c

```diff
@@ -2240,22 +2253,23 @@ static nvme_ctrl_t nvme_ctrl_alloc(nvme_root_t r, nvme_subsystem_t s,
 		errno = ENODEV;
 		return NULL;
 	}
+	FREE_CTRL_ATTR(c->address);
 	c->address = addr;
 	addr = NULL;
```

Comment:
Similar to assigning ```c->sysfs_dir``` to ```strdup(path)``` above, please assign ```c->address``` to ```strdup(addr)``` as well and remove the ```addr = NULL``` statement:

```
-       c->address = addr;
-       addr = NULL;
+       c->address = strdup(addr);
```

That makes it more intuitive and easier to read.